umbra-js/src/classes/Umbra.ts

```diff
@@ -721,3 +750,46 @@ async function tryEthWithdraw(
     return await tryEthWithdraw(signer, from, to, overrides, retryCount + 1);
   }
 }
+
+export async function assertSufficientBalance(signer: JsonRpcSigner | Wallet, token: string, tokenAmount: BigNumber) {
+  // If applicable, check that sender has sufficient token balance. ETH balance is checked on send. The isEth
+  // method also serves to validate the token input
+  if (!isEth(token)) {
+    const tokenContract = new Contract(token, ERC20_ABI, signer) as ERC20;
+    const tokenBalance = await tokenContract.balanceOf(await signer.getAddress());
+    if (tokenBalance.lt(tokenAmount)) {
+      const providedAmount = tokenAmount.toString();
+      const details = `Has ${tokenBalance.toString()} tokens, tried to send ${providedAmount} tokens.`;
+      throw new Error(`Insufficient balance to complete transfer. ${details}`);
+    }
+  }
+  return true;
+}
+
+export function assertValidStealthAddress(stealthAddress: string) {
+  // Ensure that the stealthKeyPair's address is not on the block list
+  if (blockedStealthAddresses.includes(stealthAddress)) throw new Error(`Invalid stealth address: ${stealthAddress}`);
+}
+
+/**
+ * @notice Helper method to return parsed localOverrides and LookupOverrides
+ * @param overrides Override the gas limit, gas price, nonce, or advanced mode.
+ * When `advanced` is false it looks for public keys in StealthKeyRegistry, and when true it recovers
+ * them from on-chain transaction when true
```

Comment:
I think we said we were going to add a comment here explaining why this function is exported (for testing only) and that it should only be used within this file

Reply:
Added the comment https://github.com/ScopeLift/umbra-protocol/pull/426/commits/c6ae57a55c8671619e38b1dd202e691fe4730098